facilitator/src/transport/s3.rs

```diff
@@ -668,28 +667,33 @@ mod tests {
             key: "".into(),
         };
 
-        let mut transport = S3Transport::new_with_client(s3_path.clone(), false, |region, _| {
+        let provider = Box::new(|region: &Region, _: Option<String>| {
             Ok(S3Client::new_with(
                 // Failed GetObject request
                 MockRequestDispatcher::with_status(404).with_request_checker(is_get_object_request),
                 MockCredentialsProvider,
                 region.clone(),
             ))
         });
+        let mut transport = S3Transport::new_with_client(s3_path.clone(), None, provider);
 
         let ret = transport.get(TEST_KEY);
         assert!(ret.is_err(), "unexpected return value {:?}", ret.err());
 
-        let mut transport = S3Transport::new_with_client(s3_path.clone(), false, |region, _| {
-            Ok(S3Client::new_with(
-                // Successful GetObject request
-                MockRequestDispatcher::with_status(200)
```

Comment:
This reminds me: now that we know about [mockito](https://docs.rs/mockito/0.27.0/mockito/index.html), which has proper support for verifying that mocks were consumed than what I rigged up here with Rusoto's mocking framework, we should rewrite these tests. #114